app/controllers/api/connections_controller.rb

```diff
@@ -122,10 +116,73 @@ def update_match
 
     end
 
+    def update_connection
+
+        if @connection.in_progress?
+
+            begin
+                @connection.class_status = params[:status]
+
+            rescue
+                return render_json_400("Updated status is invalid")
+            end
+
+            if @connection.in_progress?
+                return render_json_400("Updated status is invalid")
+            end
+
+            # Valid request
+
+            if @is_student
+                @connection.person_closed_connection = "student_closed"
+            else
+                @connection.person_closed_connection = "teacher_closed"
+            end
+
+            @connection.ended_at = Time.new
+
+            if @connection.save
+                # connection updated successfully, now we need to notify the other user
+
+                skill_class = @connection.match.skill_class
+
+                if @connection.given?
+                    text = "Connection to class to class #{skill_class.title} was closed. Class was given"
+                else
+                    text = "Connection to class to class #{skill_class.title} was closed. Class was canceled"
+                end
+
+
+                if @is_student
+                    # create notification for the teacher
+                    notification = Notification.new({text: text, notification_type: "connection_closed", match_id: @connection.match_id, person_id: @connection.match.student_id})
+                else
+                    #create notification for the student
+                    notification = Notification.new({text: text, notification_type: "connection_closed", match_id: @connection.match_id, person_id: skill_class.teacher_id})
+                end
+
+                unless notification.save
+                    error = {"error": notification.errors.full_messages}
+                    return render(json: error, status: 500)
+                end
+
+                res = @connection.as_json(only: [:id, :created_at, :ended_at, :class_status])
+                render(json: res)
+
+            else
+                render_json_400(@connection.errors.full_messages)
+            end
+
+        else
+            render_json_400("Connection has already been closed")
+        end
+
+    end
```

Comment:
You can reduce the nesting by doing
```ruby
return render_json_400("Connection has already been closed") unless @connection.in_progress?
```
and
```ruby
return render_json_400(@connection.errors.full_messages) unless @connection.save
```

Reply:
Noted from now on